### PR TITLE
git_create_tag doesn't have the correct descriptor

### PR DESCRIPTION
**Pull request by mattyclarkson**
The following code doesn't work:

```js
// Commit the bower.json file
.then(function() {
  return repository.createCommitOnHead(['bower.json'], identity, identity, 'release ' + version);
})

// Tag the new commit
.then(function(oid) {
  return nodegit.Commit.lookup(repository, oid);
})
.then(function(commit) {
  return nodegit.Tag.create(null, repository, version, commit, identity, 'version ' + version, 0);
})
```

This is because the `git_create_tag` needs to have the following descriptor:

```json
"git_tag_create": {
  "isAsync": true,
  "args": {
    "oid": {
    "isReturn": true,
    "isSelf": false,
    "shouldAlloc": true
  },
  "return": {
    "cppClassName": "Number",
    "jsClassName": "Number",
    "isErrorCode": true
  }
},
```

### CODE REVIEW

**Comment by mattyclarkson** [comment]
Ow. it failed tests :cry: will look through the logs.

File: test/tests/tag.js [hunk 1]
```diff
@@ -75,4 +77,67 @@ describe("Tag", function() {
...
+      })
+      .then(function() {
+        return Reference.lookup(repository, "refs/tags/" + name);
+      })
+      .then(function() {
+        return Promise.reject(new Error("the tag '" + name + "' should not exist"));
+      }, function() {
+        return Promise.resolve();
+      });
+  });
```
Comment (mattyclarkson) [inline test/tests/tag.js]: I don't really like that this test bundles both the creation and deletion together into a single test but if the tag isn't cleaned up the tests aren't repeatable.

**Comment by johnhaley81** [comment]
Looks like it failed linting. Try running `npm test` to see the linter errors.

**Comment by mattyclarkson** [comment]
huh, the linting checks on appveyor are moaning about `Promise` being a bad name. but that is used across lots of the test files. Is that something I should fix up across all the files? Not sure why that didn't show up on my box.

**Comment by mattyclarkson** [comment]
Oh, I didn't run `jshint lib test/tests examples lifecycleScripts`

**Comment by tbranyen** [comment]
Ah my mistake, can you force push a fix that doesn't use camelCase, it's all lower case per:

http://jshint.com/docs/options/#futurehostile

**Comment by mattyclarkson** [comment]
Yay! :smiley:

**Comment by johnhaley81** [comment]
:+1:

**Comment by johnhaley81** [comment]
Manually merging this

**Comment by johnhaley81** [comment]
Manually merged via https://github.com/nodegit/nodegit/commit/094f2401fed7fa3a1fdb3f673307419041acdc29

**Comment by maxkorp** [comment]
Awesome work @mattyclarkson

**Comment by mattyclarkson** [comment]
@johnhaley81, sorry I didn't merge this on friday the build failed on appveyor and I had to leave for a weekend away! Thanks for merging it in manually.

**Comment by johnhaley81** [comment]
No worries :)

**Pull request closed without merging**